submissions/OlexiyDobroskok/OOP_exercise/index.js

```diff
@@ -0,0 +1,159 @@
+const inhabitantDefaultProperties = {
+  species: "No species",
+  name: "No name",
+  gender: "No gender",
+  saying: "Hello",
+  friends: "",
+};
+
+const humanDefaultProperties = {
+  hands: 2,
+  legs: 2,
+};
+
+const petsDefaultProperties = {
+  paws: 4,
+};
+
+class Inhabitant {
+  static worldPopulation = [];
+  constructor({
+    species,
+    name,
+    gender,
+    saying,
+    friends,
+  } = inhabitantDefaultProperties) {
+    this.species = species ?? inhabitantDefaultProperties.species;
+    this.name = name ?? inhabitantDefaultProperties.name;
+    this.gender = gender ?? inhabitantDefaultProperties.gender;
+    this.saying = saying ?? inhabitantDefaultProperties.saying;
+    this.friends = friends ?? inhabitantDefaultProperties.friends;
+    Inhabitant.worldPopulation.push(this);
+  }
+
+  addFriend({ name }) {
+    this.friends += name + "; ";
+  }
+
+  introduceYourSelf() {
+    if (this.friends === "") {
+      this.friends = "Looking for friends!";
+    }
+    return `<strong>${this.saying}! My name is ${this.name}. Species: ${this.species}. Gender: ${this.gender}. Friends: ${this.friends} </strong>`;
+  }
+}
+
+class Human extends Inhabitant {
+  constructor({
+    name,
+    gender,
+    saying,
+    legs,
+    hands,
+    friends,
+  } = humanDefaultProperties) {
+    super({ species: "human", name, gender, saying, friends });
+    this.hands = hands ?? humanDefaultProperties.hands;
+    this.legs = legs ?? humanDefaultProperties.legs;
+  }
```

Comment:
Why not just to set default value the same way it is done for `species`?
This is not a rhetoric question.

Reply:
I’ve overdone the default values a little bit. =))

Reply:
Done